core/templates/domain/state/StateObjectFactory.ts

```diff
@@ -42,24 +42,30 @@ import {
 import INTERACTION_SPECS from 'interactions/interaction_specs.json';
 import constants from 'assets/constants';
 import { AppConstants } from 'app.constants';
+import { Outcome } from 'domain/exploration/OutcomeObjectFactory';
 
 export interface StateBackendDict {
-  'classifier_model_id': string;
+  // The classifier model ID associated with a state, if applicable
+  // null otherwise.
+  'classifier_model_id': string | null;
   'content': SubtitledHtmlBackendDict;
   'interaction': InteractionBackendDict;
   'param_changes': readonly ParamChangeBackendDict[];
   'recorded_voiceovers': RecordedVoiceOverBackendDict;
   'solicit_answer_details': boolean;
   'card_is_checkpoint': boolean;
   'written_translations': WrittenTranslationsBackendDict;
-  'linked_skill_id': string;
+  // This property is null if no skills are linked to an interaction.
+  'linked_skill_id': string | null;
   'next_content_id_index': number;
 }
 
+type InteractionSpecsKey = keyof typeof INTERACTION_SPECS;
```

Comment:
Redefined again.

Reply:
Updated exporting here only and using elsewhere